cli/src/main/java/com/liferay/blade/cli/util/ProductKeyInfo.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.blade.cli.util;
+
+import java.util.Comparator;
+
+/**
+ * @author Drew Brokke
+ */
+public class ProductKeyInfo implements Comparable<ProductKeyInfo> {
+
+	@Override
+	public int compareTo(final ProductKeyInfo keyInfo) {
+		return Comparator.comparing(
+			ProductKeyInfo::getProductRank
+		).thenComparing(
+			ProductKeyInfo::isQuarterly
+		).thenComparing(
+			ProductKeyInfo::getMajorProductKeyVersion
+		).thenComparing(
+			ProductKeyInfo::getMinorProductKeyVersion
+		).thenComparing(
+			ProductKeyInfo::getMicroProductKeyVersion
+		).reversed(
```

Comment:
remind me again why this is reversed?

Reply:
The way the comparators are written, the "winners" will sink to the bottom, the way that `9` would "win" over `7` and come later in the list. So this reverse puts all the winners on top instead.